File: src/main/java/org/cascadebot/cascadebot/utils/votes/VoteResult.java
/*
 * Copyright (c) 2019 CascadeBot. All rights reserved.
 * Licensed under the MIT license.
 */

package org.cascadebot.cascadebot.utils.votes;

public class VoteResult {

    private int amount;
    private Object vote;

    public VoteResult(int amount, Object vote) {
        this.amount = amount;
        this.vote = vote;
    }

    /**
     * Gets the amount of votes this got.
     *
     * @return the amount of votes this got.
     */
    public int getAmount() {
        return amount;
    }

    /**
     * Gets the unicode or {@link net.dv8tion.jda.api.entities.Emote} that was this vote.
     *
     * @return the unicode or {@link net.dv8tion.jda.api.entities.Emote} that was this vote.
     */
    public Object getVote() {
        return vote;
    }

}

File: src/main/kotlin/org/cascadebot/cascadebot/utils/votes/VoteResult.kt
/*
 * Copyright (c) 2022 CascadeBot. All rights reserved.
 * Licensed under the MIT license.
 */

package org.cascadebot.cascadebot.utils.votes

import org.cascadebot.cascadebot.data.entities.VoteType

data class VoteResult(val vote: String, val voteType: VoteType, val count: Int) : Comparable<VoteResult> {
    override fun compareTo(other: VoteResult): Int {
        val countCompare = count.compareTo(other.count)
        if (countCompare != 0) {
            return  countCompare
        }

        val voteTypeCompare = voteType.compareTo(other.voteType)
        if (voteTypeCompare != 0) {
            return voteTypeCompare
        }

        return vote.compareTo(other.vote)
    }
}